beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -2616,21 +2590,29 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                 .map_err(BeaconChainError::from)?;
         }
 
-        let mut fork_choice = self.fork_choice.write();
+        let mut fork_choice = self.canonical_head.fork_choice_write_lock();
```

Comment:
this suggestion must coincide with other suggestion below:
```suggestion
        let fork_choice = self.canonical_head.fork_choice_read_lock();
```